Lib2025/src/main/java/frc/robot/subsystems/ExtensionArm.java

```diff
@@ -0,0 +1,93 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.spark.SparkBase.ControlType;
+import com.revrobotics.spark.SparkBase.PersistMode;
+import com.revrobotics.spark.SparkBase.ResetMode;
+import com.revrobotics.spark.SparkClosedLoopController;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+import com.revrobotics.spark.SparkMax;
+import com.revrobotics.spark.config.ClosedLoopConfig;
+import com.revrobotics.spark.config.EncoderConfig;
+import com.revrobotics.spark.config.MAXMotionConfig;
+import com.revrobotics.spark.config.SparkBaseConfig;
+import com.revrobotics.spark.config.SparkMaxConfig;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Constants.EXTENSION_ARM;
+import frc.robot.util.Utility;
+
+public class ExtensionArm extends SubsystemBase {
+    private double m_targetRotations;
+
+    private SparkMax m_motor;
+    private SparkClosedLoopController m_PIDController;
+    private RelativeEncoder m_encoder;
+
+    public ExtensionArm() {
+        m_motor = new SparkMax(Constants.CAN_ID.TRAP_AMP_ARM_MOTOR_ID, MotorType.kBrushless);
+        m_PIDController = m_motor.getClosedLoopController();
+        m_encoder = m_motor.getEncoder();
+
+        configure();
+    }
+
+    private void configure() {
+        MAXMotionConfig maxMotionConfig = new MAXMotionConfig()
+                .maxVelocity(EXTENSION_ARM.SMART_MOTION_MAX_VEL_RPM)
+                .maxAcceleration(EXTENSION_ARM.SMART_MOTION_MAX_ACC_RPM)
+                .allowedClosedLoopError(EXTENSION_ARM.SMART_MOTION_ALLOWED_ERROR);
+
+        ClosedLoopConfig PIDConfig = new ClosedLoopConfig()
+                .pidf(EXTENSION_ARM.MOTOR_PID_P, EXTENSION_ARM.MOTOR_PID_I, EXTENSION_ARM.MOTOR_PID_D,
```

Comment:
We'll need to add `.feedbackSensor(FeedbackSensor.kAlternateOrExternalEncoder)` after this line